src/slow_log/slow_log.c

```diff
@@ -0,0 +1,224 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include <stdio.h>
+#include <assert.h>
+#include <unistd.h>
+
+#include "./slow_log.h"
+#include "../util/arr.h"
+#include "../query_ctx.h"
+#include "../util/rmalloc.h"
+#include "../util/thpool/thpool.h"
+
+static SlowLog *_slowlog = NULL;
+
+static int get_thread_id() {
+	extern threadpool _thpool;  // Declared in module.c
+	return thpool_get_thread_id(_thpool, pthread_self());
+}
+
+static SlowLogItem *_SlowLogItem_New(const char *cmd, const char *graph_id, const char *query,
+									 double latency) {
+	SlowLogItem *item = rm_malloc(sizeof(SlowLogItem));
+	item->cmd = rm_strdup(cmd);
+	item->query = rm_strdup(query);
+	item->graph_id = rm_strdup(graph_id);
+	item->latency = latency;
+	time(&item->time);
+	return item;
+}
+
+static void _SlowLog_Item_Free(SlowLogItem *item) {
+	assert(item);
+	rm_free(item->cmd);
+	rm_free(item->query);
+	rm_free(item->graph_id);
+	rm_free(item);
+}
+
+// Compares two heap record nodes.
+static int _slowlog_elem_compare(const void *A, const void *B, const void *udata) {
+	SlowLogItem *a = (SlowLogItem *)A;
+	SlowLogItem *b = (SlowLogItem *)B;
+	return -1 * (a->latency - b->latency);
```

Comment:
Can be written as:
`return b->latency - a->latency;`

Reply:
👍